src/main/java/com/o19s/es/ltr/ranker/dectree/ModelObjective.java

```diff
@@ -0,0 +1,59 @@
+package com.o19s.es.ltr.ranker.dectree;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Represents an XGBoost model learning objective
+ *
+ * Depending on the objective, the model prediction may require a transform. Currently only untransformed
+ * and logistic (sigmoid) types are implemented.
+ * See <a href="https://xgboost.readthedocs.io/en/latest/parameter.html#learning-task-parameters">task params</a>
+ */
+public class ModelObjective {
+    public static final String DEFAULT_OBJECTIVE_TYPE = "reg:linear";
+    private static final Map<String, ModelObjective> OBJECTIVES = new HashMap<>();
+
+    static {
+        register("binary:logistic", new LogisticObjective());
```

Comment:
I would put this map into the XGboost parser as this is really xgboost specific.